File: lib/models/state/scene_state.dart
/// Represents the state of the scene.
enum SceneState {
  /// represents idle state.
  none,

  /// represents loading state.
  loading,

  /// represents the scene has finished loading successfully.
  loaded,

  /// represents that some error happened while loading the scene.
  error;

  static SceneState from(String? state) {
    switch (state) {
      case "NONE":
        return SceneState.none;

      case "LOADING":
        return SceneState.loading;

      case "LOADED":
        return SceneState.loaded;

      case "ERROR":
        return SceneState.error;

      default:
        return SceneState.none;
    }
  }
}

File: android/src/main/kotlin/io/sourcya/playx_3d_scene/core/models/states/SceneState.kt
package io.sourcya.playx_3d_scene.core.models.states

enum class SceneState {
    NONE,
    LOADING,
    LOADED,
    ERROR;

    companion object{

        fun getSceneState(first :SceneState , second:SceneState): SceneState {
            return if(getValue(first) > getValue(second)) {
                first;
            }else{
                second
            }
        }

        fun getSceneState(first :SceneState , second:SceneState, third:SceneState): SceneState {
            val firstAndSecondState = getSceneState(first,second)
            return getSceneState(firstAndSecondState,third)
        }

        private fun getValue(state:SceneState): Int {
           return when(state){
                NONE -> 1
                LOADING -> 4
                LOADED -> 3
                ERROR -> 2
            }
        }



    }


}